### ISSUE TITLE
🐛 Destructured member from `this` are not detected in `lint/correctness/noUnusedPrivateClassMembers`

### ISSUE THREAD

**Issue by ycmjason** [biomejs/biome#7784]
### Environment information

```block
CLI:
  Version:                      2.2.5
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.9.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.18.3

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

https://biomejs.dev/playground/?lintRules=noUnusedPrivateClassMembers&code=YwBsAGEAcwBzACAAQQBuAGkAbQBhAGwAIAB7AAoAIAAgAHAAcgBpAHYAYQB0AGUAIABuAGEAbQBlADoAIABzAHQAcgBpAG4AZwA7AAoACgAgACAAZABvAFMAbwBtAGUAdABoAGkAbgBnAFcAaQB0AGgATgBhAG0AZQAoACkAIAB7AAoAIAAgACAAIABjAG8AbgBzAHQAIAB7ACAAbgBhAG0AZQAgAH0AIAA9ACAAdABoAGkAcwAKAAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAG4AYQBtAGUACgAgACAAfQAKAH0A

### Expected result

destructured members should be treated as "used"?

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by pastelmind** [comment]
Looks like a duplicate of #6974
